hypothesis-python/src/hypothesis/strategies/_internal/strings.py

```diff
@@ -159,6 +177,41 @@ def filter(self, condition):
                 f"all nonempty strings!  Did you mean str.is{condition.__name__}?",
                 HypothesisWarning,
             )
+        if condition is str.isidentifier and isinstance(
+            unwrap_strategies(self.element_strategy), OneCharStringStrategy
+        ):
+            from hypothesis.strategies._internal.core import characters, composite
+
+            @composite
+            def make_identifier(draw, element_strategy):
+                # TODO: Include Other_ID_Start_chars and Other_ID_Continue_chars
+                id_start_categories = ("Lu", "Ll", "Lt", "Lm", "Lo", "Nl")
+                id_continue_categories = id_start_categories + ("Mn", "Mc", "Nd", "Pc")
+
+                is_start_strategy = unwrap_strategies(
+                    characters(
+                        whitelist_categories=id_start_categories,
+                        whitelist_characters=("_",),
+                    )
+                ).intersection(unwrap_strategies(self.element_strategy))
```

Comment:
It's somewhat inefficient to do all this computation inside an `@composite` function, since it'd be invoked every time we went to draw an element.  Instead, let's compute `id_start_strat` and `id_continue_strat` then

```python
return st.builds("{}{}".format, id_start_strat, st.text(id_continue_strat))
```